### PR TITLE
Optionize ip-version selection in the client

### PR DESCRIPTION
**Pull request by oschaaf**
Adds a new `--address-family` option that controls behaviour of
the target host-to-address resolution process in the Nighthawk client.
Acceptable values are `[v4,v6,auto]`, defaults to `ipv4`.

https://github.com/envoyproxy/nighthawk/pull/15 needs to go in first.

### CODE REVIEW

File: test/client_test.cc [hunk 1]
```diff
@@ -93,6 +93,14 @@ class ClientTest : public Envoy::BaseIntegrationTest,
     return fmt::format("http://{}:{}/", address, port_);
   }
 
+  const char* getAddressFamilyOptionString() {
+    auto ip_version = GetParam();
+    RELEASE_ASSERT(ip_version == Envoy::Network::Address::IpVersion::v4 ||
```
Comment (htuch) [inline test/client_test.cc]: Don't we want to `throw` and print the error in the main CLI driver in a user friendly way?

File: test/options_test.cc [hunk 2]
```diff
@@ -183,6 +186,27 @@ TEST_F(OptionsImplTest, VerbosityValuesAreConstrained) {
                           MalformedArgvException, "Value 'foo' does not meet constraint");
 }
 
+/// -------
```
Comment (htuch) [inline test/options_test.cc]: ?

File: test/client_test.cc (hunk 1 above)
Comment (oschaaf) [reply test/client_test.cc]: We do so, see `ParseAddressFamilyBadValues` below. This is just for test.

Comment (oschaaf) [reply test/client_test.cc]: (this assert is primarily meant to catch when it is time for maintenance, `GetParam()` would be returning something we didn't expect.

**Comment by oschaaf** [comment]
@htuch comments addressed, ready for another look.

**Pull request merged**